ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -627,6 +633,23 @@ private JoinNode buildJoin(final Join root, final String prefix, final boolean i
     );
   }
 
+  private Optional<ColumnName> getLeftJoinNonKeyColumn(final Join join) {
+    // This method is called after the verifyJoin() has detected a FK join. This code does not
+    // check for values are in the right order. It is assumed verifyJoin() already checked them
+    // and failed in case they are not valid.
+
+    // What if expression is not ColumnReferenceExp?
+    final ColumnReferenceExp columnReference =
+        (ColumnReferenceExp) join.getInfo().getLeftJoinExpression();
+
+    final Optional<SourceName> qualifier = columnReference.maybeQualifier();
+    final ColumnName columnName = columnReference.getColumnName();
+
+    return Optional.of(qualifier.isPresent()
+        ? ColumnNames.generatedJoinColumnAlias(qualifier.get(), columnName)
```

Comment:
Why are we doing this? The column name needs to appear in the left source schema else the code in the physical later will fail. Unless I'm misunderstanding what's happening here, we should not be rewriting the column name.

Reply:
Old code. I removed this.